### ISSUE TITLE
Incorrect image _type

### ISSUE THREAD

**Issue by raffij** [ricokahler/sanity-codegen#73]
I think in `sanity-codegen.ts` the `authorThumbnail._type` should be `image`. Looks like the correct type for images are `_type="image"` for singletons and `_type=${name}` for images in arrays? 

Source schema:

```js
export default {
  title: 'Company Testimonials',
  name: 'companyTestimonials',
  type: 'object',
  icon: FaQuoteRight,
  fields: [
    {
      title: 'Company Logos',
      name: 'companyLogos',
      type: 'array',
      of: [
        {
          title: 'Company Logo',
          name: 'companyLogo',
          type: 'image',
        },
      ],
    },
    {
      title: 'Testimonials',
      name: 'testimonials',
      type: 'array',
      of: [
        {
          title: 'Testimonial',
          name: 'testimonial',
          type: 'object',
          fields: [
            {
              title: 'Image',
              name: 'image',
              type: 'image',
            },
            {
              title: 'Author Thumbnail',
              name: 'authorThumbnail',
              type: 'image',
            },
          ],
        },
      ],
    },
  ],
};
```

Vision output:

```js
{
  _key: '1814b6872bcf',
  _type: 'companyTestimonials',
  companyLogos: [
    {
      _key: '0f20b001789b',
      _type: 'companyLogo',
      asset: {
        _ref: 'image-ff5b24337398a67253510370a75db1322e27777a-1200x1600-jpg',
        _type: 'reference',
      },
    },
  ],
  testimonials: [
    {
      _key: '82faa9f96668',
      _type: 'testimonial',
      authorThumbnail: {
        _type: 'image',
        asset: {
          _ref: 'image-97cd04c5417eec077c0e385ea6b7c2531b195e51-600x600-jpg',
          _type: 'reference',
        },
      },
      image: {
        _type: 'image',
        asset: {
          _ref: 'image-bcbab495e4b32ca499c424d248811c005196385d-252x252-png',
          _type: 'reference',
        },
      },
    },
  ],
}
```

sanity-codegen.ts

```js
export type SanityCompanyTestimonialsType = {
  _type: 'companyTestimonials';
  /**
   * Company Logos — `array`
   *
   *
   */
  companyLogos?: Array<
    SanityKeyed<{
      _type: 'companyLogo';
      asset: SanityAsset;
      crop?: SanityImageCrop;
      hotspot?: SanityImageHotspot;
    }>
  >;

  /**
   * Testimonials — `array`
   *
   *
   */
  testimonials?: Array<
    SanityKeyed<{
      _type: 'testimonial';
      /**
       * Image — `image`
       *
       *
       */
      image?: {
        _type: 'image';
        asset: SanityAsset;
        crop?: SanityImageCrop;
        hotspot?: SanityImageHotspot;
      };

      /**
       * Author Thumbnail — `image`
       *
       *
       */
      authorThumbnail?: {
        _type: 'authorThumbnail';
        asset: SanityAsset;
        crop?: SanityImageCrop;
        hotspot?: SanityImageHotspot;
      };
    }>
  >;
};
```

**Comment by raffij** [comment]
#74 A crude example of a fix that works locally

**Comment by ricokahler** [comment]
Ahh interesting. Nice catch! I'll investigate and have something out before Monday.

**Comment by ricokahler** [comment]
This has been released in v0.9.0 🎉

I also added you to as a co-author so you show up in this list now:

![image](https://user-images.githubusercontent.com/10551026/107865597-28feab00-6e36-11eb-9b79-f4054abd28cc.png)

Thanks for reporting!
